tasks/download_extract_package.yml

```diff
@@ -0,0 +1,44 @@
+# SPDX-License-Identifier: MIT
+# This task checks sha256, downloads, and extracts, and then removes a tarball..
+# It takes __hpc_pkg_info dict with name, url, sha256 keys.
+# It sets __hpc_pkg_extracted variable with the path of the extracted tarball.
+---
+- name: Create temp file for downloading tarball {{ __hpc_pkg_info.name }}
+  tempfile:
+    state: file
+    prefix: "{{ __hpc_pkg_info.name }}_"
+    suffix: "{{ __hpc_pkg_info.url | basename
+      | regex_search('\\.(tar\\.gz|tbz|tar\\.bz2|zip|tar)$')
+      | default('') }}"
+  register: __hpc_pkg_tarball
+  changed_when: false
```

Comment:
Rather than having a tempfile and a temp directory to manage, could you create a single temp directory, then create files in that directory without using tempfile, then at the end you can just remove the directory which will remove all of the temp files too?

Reply:
Done